rllib/evaluation/worker_set.py

```diff
@@ -99,14 +102,49 @@ def __init__(
         }
         self._cls = RolloutWorker.as_remote(**self._remote_args).remote
         self._logdir = logdir
-
         if _setup:
             # Force a local worker if num_workers == 0 (no remote workers).
             # Otherwise, this WorkerSet would be empty.
             self._local_worker = None
             if num_workers == 0:
                 local_worker = True
-
+            if (
```

Comment:
Yeah so the story here is that I needed to replace the functionality of the old reader where in the input field, you could just specify a file path or a directory. 

I need to add all this checking so that I don't try to convert anything that isn't a file path into a dataset.

Reply:
I am still not clear why we need this. whether we are dealing with a dataset input or some other input types is not ambiguous at all right? it's specified via the "input" field. and "paths" is under "input_config" field?

Reply:
It can be ambigous though.

If I say 
```python 
# specifying a model but this is still a string
config["input"] = "ray.rllib.offline.JsonReader"

#specifying a dataset here by path, default format expected is json or a zip of a json
config["input"] = "/home/path_to_dataset"

#specifying a directory
config["input"] = "/home/dataset_directory"

#specifying a module registered with tune
tune.register("custom_reader", custom_reader)
config["input"] = "custom_reader"
```

These are all different ways of specifying an input reader via a string that we support today, but not all of them are referring to the path to a dataset directory or the direct path to a dataset, in JSON format, which is a functionality that we support today.

The if statement is filtering for the case where when a string is passed to input, if it is a path to a dataset or dataset directory then we should enter the if statement. 

IFF the string is a module path or a then we don't try to convert it to a dataset. 

This logic is technically implemented later in worker_set.py, but needs to be implemented here for the usecase that a user passes a path to a dataset(directory) so that this path can be converted to a ray dataset.